copier/vcs.py

```diff
@@ -111,15 +114,36 @@ def checkout_latest_tag(local_repo: StrOrPath, use_prereleases: OptBool = False)
 def clone(url: str, ref: OptStr = None) -> str:
     """Clone repo into some temporary destination.
 
+    Includes dirty changes for local templates by copying into a temp
+    directory and applying a wip commit there.
+
     Args:
         url:
             Git-parseable URL of the repo. As returned by
             [get_repo][copier.vcs.get_repo].
         ref:
             Reference to checkout. For Git repos, defaults to `HEAD`.
     """
+
+    src = url
+    if Path(src).is_dir():
+        is_dirty = False
+        with local.cwd(url):
+            is_dirty = bool(git("status", "--porcelain").strip())
+        if is_dirty:
+            src = tempfile.mkdtemp(prefix=f"{__name__}.dirty.")
```

Comment:
The temporary directory should probably be removed once we're done using it. Maybe by using a `with` statement and rearranging the code so that `_clone` is inside?

Reply:
great, fixed and moved the duplicate `_clone` logic to `_perform_clone`